lib/runtime/runtime.go

```diff
@@ -70,10 +70,10 @@ func EventBuffer(context context.Context, client k.Interface,
 								return err
 							}
 							metrics.TotalEventOps.Inc()
-						}
-						if !hasUpdate {
+						} else {
 							// the channel got closed, so we need to restart
 							log.Error("Kubernetes hung up on us, restarting event watcher")
```

Comment:
It might make sense to update this log as well to avoid setting expectations that the process will recover.

Reply:
Done